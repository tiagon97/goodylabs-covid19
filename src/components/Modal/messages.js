const messages = {
  statement: 'Server side error due to reached requests limit. Please wait a few seconds and refresh again.',
  error: 'ERROR',
};

export default messages;
