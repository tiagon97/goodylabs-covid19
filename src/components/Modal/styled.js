import styled from 'styled-components';
import refreshIcon from '../../assets/refresh.svg';

export const StyledModal = styled.div`
  position: fixed;
  top: 20%;
  left: 50%;
  transform: translateX(-50%);
  width: 40vw;
  height: 250px;
  background: white;
  border-radius: 30px;
  box-shadow: 0 10px 30px -10px hsla(0, 0%, 0%, 0.5);
  padding-top: 20px;
`;

export const StyledOverlayWrapper = styled.div`
  position: fixed;
  overflow: hidden;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 999;
`;

export const StyledParagraph = styled.p`
  margin: 20px;
`;

export const StyledButton = styled.div`
  position: absolute;
  bottom: 6%;
  right: 3%;
  width: 60px;
  height: 60px;
  background-image: url(${refreshIcon});
  background-size: 35px;
  background-position: 15px 50%;
  background-repeat: no-repeat;
  outline: none;

  &:hover {
    cursor: pointer;
  }
`;
