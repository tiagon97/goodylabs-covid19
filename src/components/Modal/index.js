import React from 'react';
import PropTypes from 'prop-types';
import Heading from '../Heading/index';
import messages from './messages';
import { StyledOverlayWrapper, StyledParagraph, StyledButton, StyledModal } from './styled';

const Modal = ({ error }) => (
  <StyledOverlayWrapper>
    <StyledModal>
      <Heading main>{messages.error}</Heading>
      <StyledParagraph>{messages.statement}</StyledParagraph>
      <StyledParagraph>{error.message}</StyledParagraph>
      <StyledButton onClick={() => window.location.reload()} />
    </StyledModal>
  </StyledOverlayWrapper>
);

export default Modal;

Modal.propTypes = {
  error: PropTypes.string.isRequired,
};
