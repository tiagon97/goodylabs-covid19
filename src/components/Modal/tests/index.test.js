/**
 *
 * Tests for Modal
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import Modal from '../index';

const defaultProps = {
  error: 'error',
};

describe('<Modal />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Modal {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Modal {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
