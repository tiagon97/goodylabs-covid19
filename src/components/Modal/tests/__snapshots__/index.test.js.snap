// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should render and match the snapshot 1`] = `
<div
  class="sc-AxiKw gednU"
>
  <div
    class="sc-AxirZ dUnkeb"
  >
    <h1
      class="sc-AxjAm iyJNFV"
    >
      ERROR
    </h1>
    <p
      class="sc-AxhCb qRYnP"
    >
      Server side error due to reached requests limit. Please wait a few seconds and refresh again.
    </p>
    <p
      class="sc-AxhCb qRYnP"
    />
    <div
      class="sc-AxhUy bzImaP"
    />
  </div>
</div>
`;
